src/libraries/System.Security.Cryptography.X509Certificates/tests/ExtensionsTests/AuthorityInformationAccessTests.cs

```diff
@@ -0,0 +1,267 @@
+        {
+            const string BadEntry = "\u212C is not a B";
+
+            CryptographicException ex = Assert.Throws<CryptographicException>(
+                () => new X509AuthorityInformationAccessExtension(
+                    new[] { "A", BadEntry, "C" },
+                    new[] { "D" }));
+        }
+
+        [Fact]
+        public static void BuildInvalidCAIssuer()
+        {
+            const string BadEntry = "\u212B is not an A";
+
+            CryptographicException ex = Assert.Throws<CryptographicException>(
+                () => new X509AuthorityInformationAccessExtension(
+                    new[] { "D" },
+                    new[] { "C", "B", BadEntry }));
+        }
+
+        [Fact]
+        public static void EnumerateNull()
+        {
+            using (X509Certificate2 cert = new X509Certificate2(TestFiles.MicrosoftRootCertFile))
+            {
+                X509AuthorityInformationAccessExtension aia =
+                    cert.Extensions.OfType<X509AuthorityInformationAccessExtension>().Single();
+
+                Assert.Throws<ArgumentNullException>("accessMethodOid", () => aia.EnumerateUris((string)null));
+                Assert.Throws<ArgumentNullException>("accessMethodOid", () => aia.EnumerateUris((Oid)null));
+                Assert.Throws<ArgumentException>("accessMethodOid", () => aia.EnumerateUris(new Oid(null, "potato")));
+                Assert.Throws<ArgumentException>("accessMethodOid", () => aia.EnumerateUris(new Oid("", "potato")));
+            }
+        }
+
+        [Fact]
+        public static void EnumerateOidUsesValue()
+        {
+            using (X509Certificate2 cert = new X509Certificate2(TestFiles.MicrosoftRootCertFile))
+            {
+                X509AuthorityInformationAccessExtension aia =
+                    cert.Extensions.OfType<X509AuthorityInformationAccessExtension>().Single();
+
+                string[] caIssuersValues =
+                {
+                    "ldap:///CN=Microsoft%20Corporate%20Root%20Authority,CN=AIA,CN=Public%20Key%20Services,CN=Services,CN=Configuration,DC=Corp,DC=Microsoft,DC=Com?cACertificate?base?objectclass=certificationAuthority",
+                    "ldap://corp.microsoft.com/CN=Microsoft%20Corporate%20Root%20Authority,CN=AIA,CN=Public%20Key%20Services,CN=Services,CN=Configuration,DC=Corp,DC=Microsoft,DC=Com?cACertificate?base?objectclass=certificationAuthority",
+                    "http://www.microsoft.com/pki/mscorp/mscra1.crt",
+                };
+
+                const string OcspEndpoint = "1.3.6.1.5.5.7.48.1";
+                const string CertificateAuthorityIssuers = "1.3.6.1.5.5.7.48.2";
+
+                Assert.Empty(aia.EnumerateUris(new Oid(value: OcspEndpoint, friendlyName: CertificateAuthorityIssuers)));
+                Assert.Equal(
+                    caIssuersValues,
+                    aia.EnumerateUris(new Oid(value: CertificateAuthorityIssuers, friendlyName: OcspEndpoint)));
+            }
+        }
```

Comment:
Maybe overkill, but since these methods are `IEnumerable<...>` do you think there is value in a test that creates the enumerator multiple times and they are distinct enumerators?

```
_ = aia.EnumerateOcspUris().ToArray(); // Create an enumerator and consume it
Assert.Equal(expected, aia.EnumerateOcspUris()); // Different, functional enumerator.
```